vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp

```diff
@@ -217,23 +217,26 @@ void NativeUIManager::AddRootView(ShadowNode &shadowNode, facebook::react::IReac
   SystraceSection s("NativeUIManager::AddRootView");
   auto xamlRootView = static_cast<IXamlRootView *>(pReactRootView);
   XamlView view = xamlRootView->GetXamlView();
+  const auto rootTag = shadowNode.m_tag;
   m_tagsToXamlReactControl.emplace(
-      shadowNode.m_tag,
+      rootTag,
       winrt::weak_ref<winrt::Microsoft::ReactNative::ReactRootView>(
           view.as<winrt::Microsoft::ReactNative::ReactRootView>()));
 
   // Push the appropriate FlowDirection into the root view.
   view.as<xaml::FrameworkElement>().FlowDirection(
       I18nManager::IsRTL(m_context.Properties()) ? xaml::FlowDirection::RightToLeft : xaml::FlowDirection::LeftToRight);
 
-  m_tagsToYogaNodes.emplace(shadowNode.m_tag, make_yoga_node(m_yogaConfig));
+  m_tagsToYogaNodes.emplace(rootTag, make_yoga_node(m_yogaConfig));
 
   auto element = view.as<xaml::FrameworkElement>();
-  Microsoft::ReactNative::SetTag(element, shadowNode.m_tag);
+  Microsoft::ReactNative::SetTag(element, rootTag);
 
   // Add listener to size change so we can redo the layout when that happens
-  m_sizeChangedVector.push_back(
-      view.as<xaml::FrameworkElement>().SizeChanged(winrt::auto_revoke, [this](auto &&, auto &&) { DoLayout(); }));
+  m_sizeChangedVector.push_back(view.as<xaml::FrameworkElement>().SizeChanged(
+      winrt::auto_revoke, [this, rootTag](auto &&, xaml::SizeChangedEventArgs const &args) {
```

Comment:
Is it safe to capture `rootTag`, or should it be a weak_ref? Is there a scenario where this thing gets torn down and we need to care about this SizeChanged event firing? Probably not?

I noticed that `m_sizeChangedVector` doesn't have the element removed in `removeRootView`. Preexisting, but... given that you're capturing a value now it made me wonder if that's all buttoned up.